### PR TITLE
Add readme file on using terraform

### PR DESCRIPTION
**Pull request by dividehex**
This adds some basic instructions for using terraform here.

### CODE REVIEW

File: terraform/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Using Terraform
+
+
+
+### Sub-directories and state files
+Each subdirectory under the `terraform/` directory contains an isolated set of terraform configuration files, with the expectation of a few symlinked files, and has a separate state file associated with it.  State files are stored with in S3 under the bucket titled `tfstate`.
```
Comment (nthomas-mozilla) [inline terraform/README.md]: Nit, s/with in/in/ ?
re state files, looks like the bucket is callled tf-base, with data in the tf_state prefix.

File: terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+`git clone git@github.com:mozilla-releng/build-cloud-tools.git`
+
+
+### Initial setup
+
+If you haven't done so previously, you must first run `terraform init` for each subdirectory you intend to run terraform in.  This only needs to be run once on your local subdirectory in order to setup the local state files and download any missing modules.  As the terraform documentation states, `terraform init` is idempotent so it should be safe to run multiple times under the same subdirectory
+
+The following initializes the base subdirectory
+
+    cd terraform/base`
```
Comment (nthomas-mozilla) [inline terraform/README.md]: Nit, stray `

File: terraform/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    terraform init
+
+
+### Test the current state and the master branch
+Before making any changes, it is a good idea to ensure the configuration files in the master branch match that which currently exists in AWS.  This can be done with `terraform plan`.  `terraform plan` is a non-destructive action.  It validates and parsers the terraform config files within the local directory, then queries the AWS api to determine the state of the resources defined locally.  If all the resources and their defined state exist and match the local terraform configs, there should be no changes proposed.  If there is any discrepancy, terraform will return a planned set of actions to either create, destroy or modify resources in AWS.
```
Comment (nthomas-mozilla) [inline terraform/README.md]: Nit, s/parsers/parses/

File: terraform/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    cd terraform/base
+    terraform plan
+
+##### If state doesn't match the master branch
+
+If you have the most recent copy of the git master branch and `terraform plan` shows it wants to make changes, then someone may have applied changes to production without pushing them to the masters branch.  If this happens, you will need to find who made the changes and request them to be submitted for review and merged for continuing.  Best practices dictates you should never apply changes before having changes reviewed and merged to master.
```
Comment (nthomas-mozilla) [inline terraform/README.md]: Nit, Could rewrite the last sentence as ... Best practice dictates that all changes should be reviewed and merged to master before being applied to production.

**Comment by dividehex** [comment]
I'll be honest and confess I wrote this in haste so I very much appreciate the keen eye and nits.  I'll follow up with another commit.

I'm not sure anyone else has had to run it in 'anger' yet which is probably a good thing.  I'll find a second pair of eyes though.

**Comment by garbas** [comment]
I would add what secrets are needed and where to get them.
for `terraform/base` i guess we need something like 

```console
$ export AWS_ACCESS_KEY_ID=???
$ export AWS_SECRET_ACCESS_KEY=???
```

File: terraform/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# Using Terraform
+
+
+
+### Sub-directories and state files
+Each subdirectory under the `terraform/` directory contains an isolated set of terraform configuration files, with the expectation of a few symlinked files, and has a separate state file associated with it.  State files are stored in S3 under the bucket/prefix titled `tf-base/tf_state`.
```
[the remark this answers is not in the document]
Comment (dividehex) [reply terraform/README.md]: Yes, but that can come in a separate PR.  There has also been a discussion on splitting some resources in the base state to other state files.  I think that would be a good time to add other READMEs to each.

**Pull request merged**